packages/hardhat/contracts/FundingVaultERC20.sol

```diff
@@ -142,6 +143,13 @@ contract FundingVault {
 		);
 	}
 
+    function redeem() external{
+		if(block.timestamp < timestamp) revert DeadlineNotPassed();
```

Comment:
@gurukiran7 , this is not the correct voucher logic. Have a look on what @zain171m did for the non-ERC20 version. Try to understand it and do the same. If you need help, let's discuss in Discord.